### PR TITLE
Feature/401 bdo reading

### LINKED ISSUE #401: Problem with reading BDO file from SHIELD-HIT12A simulation
Simulation was performed using development version of SH12A, branch 
`379-material-and-zone-gets-normalised`, last commit today

```
(venv) ➜  SecondaryFragments git:(main) ✗ convertmc --version      
1.6.0
(venv) ➜  SecondaryFragments git:(main) ✗ convertmc inspect sh12a/depth_profile_avgenergy_zoom.bdo 
ValueError: 43522 is not a valid SHBDOTagID

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/bin/convertmc", line 8, in <module>
    sys.exit(main())
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/run.py", line 153, in main
    status = convertfromlist(parsed_args.input,
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/input_output.py", line 178, in convertfromlist
    estimator = fromfilelist(filelist, error, nan)
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/input_output.py", line 78, in fromfilelist
    result = fromfile(input_file_list[0])
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/input_output.py", line 55, in fromfile
    if not reader.read(estimator):  # unsuccefful read
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/readers/common.py", line 23, in read
    result = self.read_data(estimator)
  File "/home/grzanka/PycharmProjects/SecondaryFragments/venv/lib/python3.8/site-packages/pymchelper/readers/shieldhit/reader_bdo2019.py", line 51, in read_data
    SHBDOTagID(token_id).name,
  File "/usr/lib/python3.8/enum.py", line 309, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib/python3.8/enum.py", line 600, in __new__
    raise exc
  File "/usr/lib/python3.8/enum.py", line 584, in __new__
    result = cls._missing_(value)
  File "/usr/lib/python3.8/enum.py", line 613, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 43522 is not a valid SHBDOTagID
```

### PR DESCRIPTION
**Pull request by grzanka**
Fixes #401

### CODE REVIEW

File: pymchelper/readers/shieldhit/reader_bdo2019.py [hunk 1]
```diff
@@ -47,41 +48,50 @@ def read_data(self, estimator):
...
+                    token_name = SHBDOTagID(token_id).name
+                    logger.debug("Read token {:s} (0x{:02x}) value {} type {:s} length {:d}".format(
+                        token_name,
+                        token_id,
+                        raw_payload,
+                        token_type.decode('ASCII'),
+                        payload_len
+                    ))
+                except ValueError:
+                    logger.info("Skipping token (0x{:02x}) value {} type {:s} length {:d}".format(
```
Comment (nbassler) [inline pymchelper/readers/shieldhit/reader_bdo2019.py:61]: maybe add why it was skipped to user?

File: pymchelper/readers/shieldhit/reader_bdo2016.py [hunk 2]
```diff
@@ -49,66 +50,75 @@ def read_data(self, estimator):
...
+                    token_name = SHBDOTagID(pl_id).name
+                    logger.debug("Read token {:s} (0x{:02x}) value {} type {:s} length {:d}".format(
+                        token_name,
+                        pl_id,
+                        _pl,
+                        _pl_type.decode('ASCII'),
+                        _pl_len
+                    ))
+                except ValueError:
+                    logger.info("Skipping token (0x{:02x}) value {} type {:s} length {:d}".format(
```
Comment (nbassler) [inline pymchelper/readers/shieldhit/reader_bdo2016.py:63]: (see below)

File: pymchelper/page.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    @property
+    def data(self):
+        """
+        3-D view of page data.
+
+        Page data is stored originally in `data_raw` 1-D array.
+        This property provides efficient view of data, suitable for numpy-like indexing.
+
+        >>> e = Estimator()
```
Comment (nbassler) [inline pymchelper/page.py:78]: nice with examples

File: pymchelper/readers/shieldhit/reader_bdo2019.py (hunk 1 above)
Comment (grzanka) [reply pymchelper/readers/shieldhit/reader_bdo2019.py:61]: What do you think about:
```
"Found unknown token (0x{:02x}) value {} type {:s} length {:d}, skipping"
```
?

Comment (grzanka) [reply pymchelper/readers/shieldhit/reader_bdo2019.py]: See https://github.com/DataMedSci/pymchelper/pull/402/commits/a8a85a2d1e749906297c1f84abf2a54c1acb7342

Comment (nbassler) [reply pymchelper/readers/shieldhit/reader_bdo2019.py]: much better, yes. thanks

**Review by nbassler — APPROVED** [review]

**Pull request merged**